### PR TITLE
pinch valve feedback control

### PR DESCRIPTION
**Pull request by mlp6**

### CODE REVIEW

**Comment by mlp6** [comment]
@DanielHull is this the only branch that would need to be merged into `master` to capture all of your pinch valve control software?

File: software/ADPL_electron/ADPL_electron.ino [hunk 1]
```diff
@@ -39,31 +39,47 @@ Pump pump(PUMP);
...
+
+
+#include "PinchValve.h"
+PinchValve pinchValve(DIR, STEP, SLEEP, UP, DOWN, RESET);
+#define FEEDBACK_RESOLUTION 0.125 // mm of movement 16/turn
+#define PUSH_BUTTON_RESOLUTION 1.0 // mm of movement
+#define HALF_RESOLUTION 0.5 // mm of movement
+#define UNCLOG_RESOLUTION 4.0 // mm of movment
+#define MAX_POSITION 5.0 // in mm
+#define MIN_POSITION 0.0 // in mm
```
Comment (mlp6) [inline software/ADPL_electron/ADPL_electron.ino:54]: Are these define constants expected to change for each site, or be something somewhat "fixed" for each pinch valve deployment?  If fixed, then I'm inclined to make these private constants in the `PinchValve` class.

File: software/ADPL_electron/ADPL_electron.ino [hunk 2]
```diff
@@ -95,6 +111,50 @@ void loop() {
...
+
+    currentTime = millis(); // clog handle, if there hasn't been a tip in a long while open all the way up and come back to optimum
+    if ((currentTime-bucket.lastTime)>(2*bucket.lowFlow)) {
+      pinchValve.shiftUp(UNCLOG_RESOLUTION);
+      pinchValve.shiftDown(UNCLOG_RESOLUTION);
+      pinchValve.shiftUp(UNCLOG_RESOLUTION);
+      pinchValve.shiftDown(UNCLOG_RESOLUTION);
+      bucket.lastTime = currentTime;
+      pinchValve.clogCounting += 1;
+
```
Comment (mlp6) [inline software/ADPL_electron/ADPL_electron.ino]: Thoughts about making an `unclog()` method?

**Review by mlp6 — COMMENTED** [review]
Want to check on some things before merging this in; see embedded comments.

File: software/ADPL_electron/ADPL_electron.ino (hunk 1 above)
Comment (DanielHull) [reply software/ADPL_electron/ADPL_electron.ino:54]: These were hard coded in unfortunately and will likely have to change by site and as we learn more about the system.

File: software/ADPL_electron/ADPL_electron.ino (hunk 2 above)
Comment (DanielHull) [reply software/ADPL_electron/ADPL_electron.ino]: Yea that would work too!

File: software/ADPL_electron/ADPL_electron.ino (hunk 1 above)
Comment (mlp6) [reply software/ADPL_electron/ADPL_electron.ino:54]: Hmmm... okay.  We'll need to figure out a good way to let folks on site set these parameters.

Comment (mlp6) [reply software/ADPL_electron/ADPL_electron.ino:54]: I am going to leave this "as is" for now.

File: software/ADPL_electron/ADPL_electron.ino (hunk 2 above)
Comment (mlp6) [reply software/ADPL_electron/ADPL_electron.ino]: commit bf9ed99

**Pull request merged**